src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -658,6 +661,71 @@ bool OutputStateMachineEngine::_IntermediateSpaceDispatch(const wchar_t wchActio
     return fSuccess;
 }
 
+// Routine Description:
+// - Handles actions that have an intermediate '#' (0x23), such as XTPUSHSGR, XTPOPSGR
+// Arguments:
+// - wch - Character to dispatch.
+// Return Value:
+// - True if handled successfully. False otherwise.
+bool OutputStateMachineEngine::_IntermediateHashDispatch(const wchar_t wchAction,
+                                                         _In_reads_(cParams) const unsigned short* const rgusParams,
+                                                         const unsigned short cParams)
+{
+    bool fSuccess = false;
+
+    DispatchTypes::GraphicsOptions rgPushPopParams[StateMachine::s_cParamsMax];
+    size_t cOptions = StateMachine::s_cParamsMax;
+    // Ensure that there was the right number of params
+    switch (wchAction)
+    {
+    case VTActionCodes::XT_PushSgr:
+    case VTActionCodes::XT_PushSgrAlias:
+        fSuccess = _GetTypedParams(rgusParams, cParams, rgPushPopParams, &cOptions);
+        break;
+
+    case VTActionCodes::XT_PopSgr:
+    case VTActionCodes::XT_PopSgrAlias:
+        if (cParams > 0)
+        {
+            // Can't supply params for XTPOPSGR.
+            fSuccess = false;
+        }
+        else
+        {
+            fSuccess = true;
+        }

```

Comment:
Couldn't we make this whole block:
```c++
        if (cParams == 0)
        {
            // Can't supply params for XTPOPSGR.
            fSuccess = true;
        }
        break;
```
Since it'll already be false?

Reply:
sure